core/domain/story_domain.py

```diff
@@ -1040,7 +1057,10 @@ def update_node_exploration_id(
         if node_index is None:
             raise ValueError(
                 'The node with id %s is not part of this story.' % node_id)
-        if self._check_exploration_id_already_present(new_exploration_id):
+        if (
+                new_exploration_id != '' and
```

Comment:
I don't understand why we have an empty string floating around all over the place. If someone deletes the exploration in the frontend we should be passing None to the backend, not '', right? (We should forbid '' ever being the value for one of these fields -- that is not conceptually correct.)

Reply:
Done, removed empty string for exploration ID and added validation against it.